### PR TITLE
Add custom text for the donate banner

### PR DESCRIPTION
**Pull request by smcalilly**
For issue #514.

Also fixed something in the readme.

## Notes
I'm using the jinja `safe` filter in combination with the `bleach` library for sanitizing the user-input html. `bleach` lets us whitelist any html tags we want to allow. The `safe` filter then parses the html string into html. Conversation in issue #514 has more details.

## Demo
See this screenshot, with the marked-up text and the unwanted <script> tag. If the script tag was whitelisted with bleach, then that code would execute in the console instead of being added to the page. See also the source code screenshot, where the script tag is sanitized.
Screen Shot 2021-08-18 at 3 15 26 PM
Screen Shot 2021-08-20 at 9 07 03 AM

## Testing
- checkout `frontend-changes` branch locally (needs better name - sorry!)
- in the admin interface, 
  - add the `PAYROLL_SHOW_DONATE_BANNER` setting and set it to true
  - add the `DONATE_MESSAGE` and input some html with `<p>` and `<strong>` tags. It should show up.

### CODE REVIEW

File: templates/jinja2/index.html [hunk 1]
```diff
@@ -20,17 +20,7 @@
         <strong>Dear BGA readers,</strong>
```
Comment (fgregg) [inline templates/jinja2/index.html:20]: if we can use html throughout, is there a reason to not include this in the message?

**Review by fgregg — CHANGES_REQUESTED** [review]

File: templates/jinja2/index.html (hunk 1 above)
Comment (fgregg) [reply templates/jinja2/index.html:20]: why would it prevent any extra styling? we can't have a "class" be set in the user input html?

Comment (smcalilly) [reply templates/jinja2/index.html:20]: @fgregg we can very easily, i'm just not sure the technical level of whoever will be inputting this code? do they know bootstrap?

Comment (fgregg) [reply templates/jinja2/index.html:20]: they probably don't know bootstrap, but we can tell them that the "BGA reader" section should have a class in it, and they can follow that.

Comment (smcalilly) [reply templates/jinja2/index.html]: @fgregg removed the greeting with this commit: c982b43. would you want to do that with the donate button, too?

File: templates/jinja2/index.html [hunk 2]
```diff
@@ -16,21 +16,7 @@
...
-        <strong>But we need your help to keep this important source of information going.</strong> This database costs money and time to complete. We contract with outside organizations, and BGA staffers spend more than a year requesting, compiling, organizing and checking data from hundreds of government bodies across Illinois to bring this site to you for free. We don't run ads and we are a small nonprofit.
-      </p>
-
-      <p>
-        If every person who visited this site gave us $1, we could complete our fundraising for the year right now. If you find this database valuable, please take just a few minutes to support it. Thank you.
-      </p>
+      {{ donate_message|safe }}
 
       <p class="text-center">
         <a href="https://donate.bettergov.org/?utm_source=database&utm_medium=other" class="btn btn-primary btn-lg" target="_blank">
```
Comment (fgregg) [inline templates/jinja2/index.html:22]: same with the button, too i think @smcalilly

**Review by fgregg — CHANGES_REQUESTED** [review]

File: templates/jinja2/index.html (hunk 2 above)
Comment (smcalilly) [reply templates/jinja2/index.html]: @fgregg changed here: 6ad320a

**Review by fgregg — CHANGES_REQUESTED** [review]
@smcalilly

can you please add two tests.

1. that if the appropriate settings are set, then the rendered html contains the write fragment. Use the current html as the test html for the setting.
2. that if html that has not been whitelisted is added it does not render

let me know if you need help writing this type of test.

**Comment by smcalilly** [comment]
@fgregg added some tests.

**Comment by smcalilly** [comment]
@fgregg for whatever reason, the migration wasn't detected. so i recreated the database and now i can run the data migration. the boolean for showing the banner already exists in the database, so i didn't include that in the migration. see this output:
```bash
~/code/bga-payroll(frontend-changes*) » docker-compose exec app python manage.py migrate
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, data_import, extra_settings, payroll, salsa_auth, sessions
Running migrations:
  Applying payroll.0033_auto_20210827_1548...Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.IntegrityError: duplicate key value violates unique constraint "extra_settings_setting_name_key"
DETAIL:  Key (name)=(PAYROLL_SHOW_DONATE_BANNER) already exists.
```


## found a bug
i found a bug. when the banner is turned on and the page loads, the cursor focuses on the search input. this causes the page to load at a weird positioning:
Screen Shot 2021-08-30 at 9 55 26 AM

do you want me to fix that?

**Comment by smcalilly** [comment]
fixed a migration conflict, hopefully build won't fail now

**Review by fgregg — APPROVED** [review]

File: payroll/migrations/0034_auto_20210830_1459.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        <p><strong>But we need your help to keep this important source of information going.</strong> This database costs money and time to complete. We contract with outside organizations, and BGA staffers spend more than a year requesting, compiling, organizing and checking data from hundreds of government bodies across Illinois to bring this site to you for free. We don't run ads and we are a small nonprofit.</p>
+        <p>If every person who visited this site gave us $1, we could complete our fundraising for the year right now. If you find this database valuable, please take just a few minutes to support it. Thank you.</p>
+        <p class="text-center"><a class="btn btn-primary btn-lg" href="https://donate.bettergov.org" target="_blank">Please support this work today!</a></p>
+        """
+
+    donate_message = Setting(name='DONATE_MESSAGE',
+                             value_type=Setting.TYPE_TEXT,
+                             value_text=donate_message_html)
+
+    donate_message.save()
```
Comment (fgregg) [inline payroll/migrations/0034_auto_20210830_1459.py:20]: add PAYROLL_SHOW_DONATE_BANNER as a get_or_create

**Review by fgregg — CHANGES_REQUESTED** [review]

File: payroll/migrations/0034_auto_20210830_1459.py (hunk 3 above)
Comment (smcalilly) [reply payroll/migrations/0034_auto_20210830_1459.py:20]: i don't know if a `get_or_create` is possible with the extra_settings plugin. i got this in the shell:

```python
>>> s = Setting.get_or_create(name='test setting', value_type=Setting.TYPE_TEXT, value_text='it is raining')
Traceback (most recent call last):
  File "<console>", line 1, in <module>
AttributeError: type object 'Setting' has no attribute 'get_or_create'
```

going to investigate the plugin

Comment (hancush) [reply payroll/migrations/0034_auto_20210830_1459.py:20]: @smcalilly `Setting.objects.get_or_create`

Comment (smcalilly) [reply payroll/migrations/0034_auto_20210830_1459.py:20]: oh duh, i'm missing the `Setting.objects` part

Comment (smcalilly) [reply payroll/migrations/0034_auto_20210830_1459.py:20]: haha jinx @hancush

**Comment by smcalilly** [comment]
added the get or create function. ebda359

one thing to note, i consistently have to tear down the database and restore it before this migration is detected. otherwise it's not detected when i run the migrate command. i have no idea why, but it's something that we might need to deal with when we deploy to production?

i did:
```bash
docker-compose run app python manage.py migrate
Creating bga-payroll_app_run ... done
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, data_import, extra_settings, payroll, salsa_auth, sessions
Running migrations:
  No migrations to apply.
```

then:
- `docker-compose down --volumes`
- `docker-compose up postgres`
- `docker exec -i bga-payroll-postgres pg_restore -U postgres -Fc -O -v -d bga_payroll < bga_payroll.dump`

and finally, successfully:
```bash
docker-compose run app python manage.py migrate 
Creating bga-payroll-redis ... done
Creating bga-payroll_app_run ... done
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, data_import, extra_settings, payroll, salsa_auth, sessions
Running migrations:
  Applying payroll.0033_employerhighestsalaries... OK
  Applying payroll.0034_auto_20210830_1459... OK
```

note to self, don't merge until hannah's ok

**Comment by fgregg** [comment]
let's pair together to look at the migrate thing.

**Review by fgregg — APPROVED** [review]

**Comment by hancush** [comment]
@smcalilly If you change a migration you already applied, Django isn't smart enough to re-apply it. You have to migrate backward to the previous migration and then re-apply the updated migration:

```bash
docker-compose run --rm app python manage.py migrate ${YOUR_APP} ${PREVIOUS_MIGRATION_NUMBER}
docker-compose run --rm app python manage.py migrate
```

**Pull request merged**